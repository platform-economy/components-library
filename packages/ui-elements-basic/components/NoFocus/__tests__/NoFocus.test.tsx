import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import * as enzyme from 'enzyme';
import 'jest-styled-components';

enzyme.configure({ adapter: new Adapter() });

import { NoFocus } from '../index';

describe('ui-elements-basic', () => {
  describe('Components', () => {
    describe('NoFocus', () => {
      it('should properly render with another element inside', () => {
        const tree = TestRenderer
          .create(<NoFocus><button /></NoFocus>)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
      it('should stop onClick event propagation', () => {
        let counter = 0;
        const increment = () => { counter += 1; };
        const element = enzyme.mount(
          <div onClick={increment}>
            <div>
              <button id="propagating-button">Button</button>
            </div>
            <NoFocus>
              <button id="not-propagating-button">Button</button>
            </NoFocus>
          </div>,
          );
        element.find('#propagating-button').simulate('click');
        expect(counter).toBe(1);
        element.find('#not-propagating-button').simulate('click');
        expect(counter).toBe(1);
      });
    });
  });
});
