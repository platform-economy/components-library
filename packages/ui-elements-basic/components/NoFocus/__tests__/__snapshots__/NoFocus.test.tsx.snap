// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-basic Components NoFocus should properly render with another element inside 1`] = `
<div
  className="nofocus"
  onClick={[Function]}
>
  <button />
</div>
`;
